lib/periodicChecks.js

```diff
@@ -0,0 +1,213 @@
+          'pending comments before merge. Thanks!',
+      })
+    );
+  } else {
+    // Assign project owner.
+    const label = getChangelogLabelFromPullRequest(pullRequest);
+    if (label) {
+      const projectOwner = getProjectOwnerFromLabel(label);
+      await context.github.issues.addAssignees(
+        context.repo({
+          issue_number: pullRequest.number,
+          assignees: [projectOwner],
+        })
+      );
+      await context.github.issues.createComment(
+        context.repo({
+          issue_number: pullRequest.number,
+          body:
+            'Hi @' +
+            projectOwner +
+            ', this PR is ready to be merged. Please make sure there are no ' +
+            "pending comments from the author's end before merge. Thanks!",
+        })
+      );
+    } else {
+      await assignPrAuthor(context, pullRequest);
+      await context.github.issues.createComment(
+        context.repo({
+          issue_number: pullRequest.number,
+          body:
+            'Hi @' +
+            pullRequest.user.login +
+            ', this PR is ready to be merged. Please ask one of the ' +
+            'reviewers to help with the merge, also, make sure there are no ' +
+            'pending comments before merge. Thanks!',
+        })
+      );
+    }
+  }
+};
+
+/**
+ * This function ensures that a pull request is assigned.
+ * 1. Fetch all pull requests.
+ * 2. Filter out those without assignees.
+ * 3. Find out appropriate assignee.
+ * 4. Assign appropriate assignee.
+ * 5. Ping appropriate assignee.
+ * @param {import('probot').Context} context
+ */
+const ensurePullRequestIsAssigned = async (context) => {
+  const allOpenPullRequests = await getAllOpenPullRequests(context);
+  const unassignedPullRequests = allOpenPullRequests.filter((pullRequest) => {
+    return pullRequest.assignees.length === 0;
+  });
+
+  unassignedPullRequests.forEach(async (pullRequest) => {
+    const pullRequestResponse = await context.github.pulls.get(
+      context.repo({
+        pull_number: pullRequest.number,
+      })
+    );
+    const pullRequestData = pullRequestResponse.data;
+
+    const hasBeenApproved = await hasPullRequestBeenApproved(
+      context,
+      pullRequestData.number
+    );
+    const hasChangesRequested = await doesPullRequestHaveChangesRequested(
+      context,
+      pullRequestData.number
+    );
+
+    if (hasMergeConflict(pullRequestData)) {
+      console.log('PULL REQUEST HAS MERGE CONFLICT...');
+      await mergeConflictModule.checkMergeConflictsInPullRequest(
+        context,
+        pullRequestData
+      );
+    } else if (hasPendingReviews(pullRequestData)) {
+      console.log('PULL REQUEST HAS PENDING REVIEWS...');
+      await pingAndAssignReviewers(context, pullRequestData);
+    } else if (hasChangesRequested) {
+      console.log('PULL REQUEST HAS CHANGES REQUESTED...');
+      await assignPrAuthor(context, pullRequestData);
```

Comment:
Why not pingAndAssignAuthor? Pass the comment to the function if it is different for different usecases.